parser/src/main/java/log/parser/service/impl/MatchServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package log.parser.service.impl;
+
+import java.security.SecureRandom;
+import java.util.ArrayList;
+
+import log.parser.service.MatchService;
+
+public class MatchServiceImpl implements MatchService {
+
+//	@Autowired
+//	private MatchRepositoryImpl repo;
+	
+	
+	@Override
+	public Long ingestMatch(String payload) {
+		SecureRandom random = new SecureRandom();
+		long matchId = Math.abs(random.nextLong());
+		String[] lineList = payload.split("\\r?\\n");
```

Comment:
The functionality of parsing the text file belongs to a MatchParser class with the following interface:

```
interface MatchParser {
    Match parse(string fileContents);
}
```

Where the returned Match object will contain all the associated kills, buys, casts, etc. MatchServiceImp should then parse each of them correspondingly.